### PR TITLE
[CNFT2-1582] - Search Value set

### PR DESCRIPTION
**Pull request by lramakr**
## Description

Developed Search Value set

## Tickets

* [Jira Ticket] https://cdc-nbs.atlassian.net/browse/CNFT2-1582

## Checklist before requesting a review
- [x] PR focuses on a single story
- [x] Code has been fully tested to meet acceptance criteria
- [x] PR is reasonably small and reviewable (Generally less than 10 files and 500 changed lines)
- [x] All new functions/classes/components reasonably small
- [x] Functions/classes/components focused on one responsibility
- [x] Code easy to understand and modify (clarity over concise/clever)
- [x] PRs containing TypeScript follow the [Do's and Don'ts](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html)
- [x] PR does not contain hardcoded values (Uses constants)
- [x] All code is covered by unit or feature tests

### CODE REVIEW

**Comment by mpeels** [comment]
1. Has the design team signed off on the existing `ValueSetLibrary` component?
2. Similar to the `QuestionLibrary`, I was expecting it to only appear in a modal and therefore not need a route. Is this correct?

**Comment by lramakr** [comment]
Yes Mike. 
There are 2 scenario. 
1. The question Library opens from left side nav where you dont see the checkboxes to select that row and add to page where it opens in the page itself
2. The question library opens from Add Question on edit page and it has the check boxes to select it to add to page and this open in the modal. I have added the route for this scenario.

**Comment by mpeels** [comment]
> Yes Mike. There are 2 scenario.
> 
> 1. The question Library opens from left side nav where you dont see the checkboxes to select that row and add to page where it opens in the page itself
> 2. The question library opens from Add Question on edit page and it has the check boxes to select it to add to page and this open in the modal. I have added the route for this scenario.

Scenario #1 is not currently in scope. The left side nav always routes to Classic NBS. First the first iteration of Page Builder only scenario 2 is valid.

File: apps/modernization-ui/src/apps/page-builder/pages/ValuesetLibrary/ValuesetLibraryTable.tsx
Comment (alaapbharadwaj) [inline apps/modernization-ui/src/apps/page-builder/pages/ValuesetLibrary/ValuesetLibraryTable.tsx:1]: Issue(blocking): in line 41 an authorization const is created, could use the `authorization()` that is imported in

File: apps/modernization-ui/src/routes/AppRoutes.tsx [hunk 1]
```diff
@@ -91,8 +92,9 @@ export const AppRoutes = () => {
                                 {config.features.pageBuilder.enabled && (
                                     <Route path="/page-builder" element={<PageBuilderContextProvider />}>
                                         <Route path="pages">
-                                            {pageLibraryRoutes(config.features.pageBuilder.page.library.enabled)}
-                                            {pageManagementRoutes(config.features.pageBuilder.page.management.enabled)}
+                                            <Route path="valueset-library" element={<ValuesetLibrary />} />
```
Comment (mpeels) [inline apps/modernization-ui/src/routes/AppRoutes.tsx:95]: Issue: Value set library should only be accessible from within a modal during the create question flow.

File: apps/modernization-ui/src/apps/page-builder/pages/ValuesetLibrary/useValuesetAPI.ts [hunk 2]
```diff
@@ -23,7 +23,11 @@ export const fetchValueSet = (
         return ValueSetControllerService.searchValueSetUsingPost({
             authorization,
             search: search
-                ? { valueSetNm: search, valueSetDescription: search, valueSetTypeCd: filter?.questionType }
+                ? {
+                      valueSetNm: search,
+                      valueSetDescription: search
+                      // valueSetCode: ''
```
Comment (alaapbharadwaj) [inline apps/modernization-ui/src/apps/page-builder/pages/ValuesetLibrary/useValuesetAPI.ts:29]: Question: Will this comment be used in the future? Or can it be removed?

File: apps/modernization-ui/src/apps/page-builder/generated/models/ValueSet.ts [hunk 3]
```diff
@@ -28,5 +28,8 @@ export type ValueSet = {
     valueSetStatusCd?: string;
     valueSetStatusTime?: string;
     valueSetTypeCd?: string;
+    valueSetDescription?: string;
+    type?: string;
+    valueSetName?: string;
```
Comment (mpeels) [inline apps/modernization-ui/src/apps/page-builder/generated/models/ValueSet.ts:33]: Question: I see changes to a `generated` file but no changes to the API. Was this re-generated based on changes from another PR or were these changes applied manually?

**Review by alaapbharadwaj — CHANGES_REQUESTED** [review]

**Review by mpeels — CHANGES_REQUESTED** [review]

**Comment by mpeels** [comment]
temporarily closing this PR as this has been de-prioritized until completion of all create question functionality

**Pull request closed without merging**